controller/apps.go

```diff
@@ -61,36 +59,43 @@ func getAndCheckOSIOClientV1(ctx context.Context) *OSIOClientV1 {
 	if os.Getenv("FABRIC8_WIT_API_URL") != "" {
 		witurl, err := url.Parse(os.Getenv("FABRIC8_WIT_API_URL"))
 		if err != nil {
-			log.Warn(ctx, nil, "Cannot parse FABRIC8_WIT_API_URL; assuming localhost")
+			return nil, errs.Wrapf(err, "cannot parse FABRIC8_WIT_API_URL")
 		}
 		host = witurl.Host
 		scheme = witurl.Scheme
 	}
 
 	oc := NewOSIOClientV1(ctx, scheme, host)
 
-	return oc
+	return oc, nil
 }
 
 func (c *AppsController) getSpaceNameFromSpaceID(ctx context.Context, spaceID uuid.UUID) (*string, error) {
 	// TODO - add a cache in AppsController - but will break if user can change space name
 	// use WIT API to convert Space UUID to Space name
-	osioclient := getAndCheckOSIOClientV1(ctx)
+	osioclient, err := getAndCheckOSIOClientV1(ctx)
+	if err != nil {
+		return nil, err
+	}
 
 	osioSpace, err := osioclient.GetSpaceByID(ctx, spaceID)
 	if err != nil {
-		return nil, err
+		return nil, errs.Wrapf(err, "unable to convert space UUID %s to space name", spaceID)
 	}
 	return osioSpace.Attributes.Name, nil
 }
 
 func getNamespaceNameV1(ctx context.Context) (*string, error) {
 
-	osioclient := getAndCheckOSIOClientV1(ctx)
-	kubeSpaceAttr, err := osioclient.GetNamespaceByType(ctx, nil, "user")
+	osioclient, err := getAndCheckOSIOClientV1(ctx)
 	if err != nil {
 		return nil, err
 	}
+
+	kubeSpaceAttr, err := osioclient.GetNamespaceByType(ctx, nil, "user")
+	if err != nil {
+		return nil, errs.Wrap(err, "unable to retrieve 'user' namespace")
+	}
 	if kubeSpaceAttr == nil || kubeSpaceAttr.Name == nil {
```

Comment:
Why the error and value check.. Shouldn't GetNamespaceByType return a err if the data is not valid?

Reply:
In this use case it could be coded that way.  I will make it return an error if if finds a namespace with no name.
GetNamespaceByType() could be called on a non-existent namespace, so returning a nil is valid, I think.

Reply:
@stooke but why not issue a NotFoundError when querying for a non-existing namespace?

Reply:
(I'm not sure why this is still showing apps.go instead of deployments.go, but the code is more or less the same)

The code returns a NotFoundError if there was no such namespace of the given type, but will return other error types if, for example there was an HTTP error calling the API or an issue unmarshalling the response JSON.  The underlying GetNamespaceByType() doesn't treat this as an error because it doesn't assume the use case enforced by this method.